File: PHP/Solid Principles/open_close/problem/PaymentProcessor.php
<?php

use Solid\HelperClasses\Payment;
use Solid\HelperClasses\Order;

class PaymentProcessor {

    public function processPayment(Order $order, Payment $payment) {
        echo "Processing payment of order {$order->getName()}\n";
        echo "Issuing payment for amount {$order->getPrice()}\n";

        switch ($payment->getType()) {
            case 'VISA':
                echo "Processing visa card payments...\n";
                break;
            case 'Master_Card':
                echo "Processing master card payments...\n";
                break;
            case 'American_Express':
                echo "Processing american express card payments...\n";
                break;
            default:
                throw new Exception("Unsupported payment type: {$payment->getType()}\n");
        }
    }
}

File: Python/SOLID/single_res/solution/PaymentProcessor.py
from order import Order
from payment import Payment


class PaymentProcessor:
    def process_payment(self, order: Order, payment: Payment):
        print(f"Processing payment of order {order.get_name()}")
        print(f"Issuing payment for amount {order.get_price()}")

        payment_type = payment.get_type()
        if payment_type == 'VISA':
            print("Processing VISA card payments...")
        elif payment_type == 'Master_Card':
            print("Processing MasterCard payments...")
        elif payment_type == 'American_Express':
            print("Processing American Express card payments...")
        else:
            raise Exception(f"Unsupported payment type: {payment_type}")
